src/Microsoft.Azure.SignalR.Common/Endpoints/AadAccessKey.cs

```diff
@@ -41,13 +68,43 @@ public async Task AuthorizeAsync(string endpoint, int? port, string serverId, Ca
             return await base.GenerateAccessToken(audience, claims, lifetime, algorithm);
         }
 
-        public Task<string> GenerateAadToken()
+        public async Task UpdateAccessKeyAsync(IServerNameProvider provider, ServiceEndpoint endpoint, ILoggerFactory loggerFactory)
         {
-            if (_authOptions is IAadTokenGenerator options)
+            if (!_initializeTcs.TrySetResult(true))
             {
-                return options.AcquireAccessToken();
+                return;
+            }
+
+            _timer.Start();
+
+            var logger = loggerFactory.CreateLogger<AadAccessKey>();
+
+            while (await _timer)
+            {
+                var isAuthorized = false;
+                for (int i = 0; i < AuthorizeMaxRetryTimes; i++)
+                {
+                    logger.LogInformation($"Try authorizing AccessKey...({i})");
+                    var source = new CancellationTokenSource(AuthorizeTimeout);
+                    try
+                    {
+                        await AuthorizeAsync(endpoint.Endpoint, endpoint.Port, provider.GetName(), source.Token);
+                        logger.LogInformation("AccessKey has been authorized successfully.");
```

Comment:
For logs, please follow such coding convention https://github.com/Azure/azure-signalr/blob/b5a82860e9af41d23f582a6c87522fb30318596f/src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpointManagerBase.cs#L333